tests/workflows/test_main_snapshot.py

```
@@ -0,0 +1,178 @@
+    def test_real_manifest_output_structure(self, real_manifest: Path) -> None:
+        """TSV output for the real manifest has three non-empty columns per line."""
+        if not real_manifest.exists():
+            pytest.skip("real manifest not available")
+
+        result = _collect_manifest_lines(real_manifest)
+
+        match result:
+            case list():
+                pass
+            case _:
+                raise AssertionError(f"expected list, got {type(result)}")
```

Ruff (0.15.9)

[warning] Avoid specifying long messages outside the exception class

(TRY003)